src/Configurator/DockerComposeConfigurator.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Flex\Configurator;
+
+use Symfony\Flex\Lock;
+use Symfony\Flex\Recipe;
+
+/**
+ * Adds services and volumes to docker-compose.yaml file.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class DockerComposeConfigurator extends AbstractConfigurator
+{
+    public function configure(Recipe $recipe, $config, Lock $lock, array $options = [])
+    {
+        $installDocker = $this->composer->getPackage()->getExtra()['symfony']['docker'] ?? false;
+        if (!$installDocker) {
+            return;
+        }
+
+        $rootDir = $this->options->get('root-dir');
+        if (
+            (
+                !file_exists($dockerCompose = $rootDir.'/docker-compose.yml') &&
+                !file_exists($dockerCompose = $rootDir.'/docker-compose.yaml')
+            ) || $this->isFileMarked($recipe, $dockerCompose)
+        ) {
+            return;
+        }
+
+        $this->write('Adding Docker Compose entries');
+
+        $offset = 8;
+        $node = null;
+        $endAt = [];
+        $lines = [];
+        foreach (file($dockerCompose) as $i => $line) {
+            $lines[] = $line;
+            $ltrimedLine = ltrim($line, ' ');
+
+            // Skip blank lines and comments
+            if (('' !== $ltrimedLine && 0 === strpos($ltrimedLine, '#')) || '' === trim($line)) {
+                continue;
+            }
+
+            // Extract Docker Compose keys (usually "services" and "volumes")
```

Comment:
Shouldn't it use the Symfony YAML component to properly parse the YAML file? Or is it really undesirable to add an additional dependency?

Reply:
AFAIK, it's not desirable to add such dependency in a Composer plugin.

Reply:
But we could use it if it's available?!

Reply:
Having different behaviors regarding the installed dependencies will be a pain to debug, isn't it? Maybe could we use @theofidry's PHP Scoper to ship our custom version of the component, but it could we handle that in another PR? Existing YAML manipulation is already done this way, if we change, we must change everywhere.